packages/expo-notifications/src/scheduleNotificationAsync.ts

```diff
@@ -191,3 +190,67 @@ function isSecondsPropertyMisusedInCalendarTriggerInput(
       Object.keys(triggerWithoutChannelId).length > 1)
   );
 }
+
+function validateDateComponentsInTrigger(
+  trigger: NonNullable<NotificationTriggerInput>,
+  components: string[]
+) {
+  const anyTriggerType = trigger as any;
+  components.forEach(component => {
+    if (!(component in anyTriggerType) || typeof anyTriggerType[component] !== 'number') {
+      throw new TypeError(`Parameter ${component} needs to be a number`);
+    }
```

Comment:
I think if the component is missing from the trigger type, we should throw an error saying so (rather than `${component} needs to be a number`)

I also don't think we need `typeof anyTriggerType[component] !== 'number'` here (since before, using `day: '5'` would've worked, but with the addition of that gate it would fail)

Reply:
ok regarding the first comment, I will separate the check in two then, one for existence (`in`) and the other one about type.

But regarding the second, are you sure? Because all the XXXTriggerInput are defined with date components of type `number`, so TypeScript should complain if you try to send a string… see thee screenshot:

![image](https://user-images.githubusercontent.com/632782/108258550-e80cdc00-713e-11eb-845f-e2f17cffc60e.png)

I don't mind removing that check, but in my opinion it's compatible with how the types are currently declared.

Reply:
Actually- since the native code is expecting a `Number` both on iOS and Android, this is a good change and we should keep it 👍  thanks for pushing back there